packages/domain-picker/test/search-and-select-domain-suggestion.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { DataStatus, DomainSuggestion } from '@automattic/data-stores/src/domain-suggestions';
+
+/**
+ * Internal dependencies
+ */
+import '../matchMedia.mock';
+import DomainPicker from '../src/domain-picker';
+import { useDomainSuggestions } from '../src/hooks/use-domain-suggestions';
+
+// TODO: should we extract to separate mock data file/folder?
+const MOCK_SUGGESTION: DomainSuggestion = {
+	domain_name: 'example.com',
+	cost: '€15.00',
+	raw_price: 15,
+	currency_code: 'EUR',
+	relevance: 0.5,
+};
+
+const mockUseDomainSuggestionsResult: ReturnType< typeof useDomainSuggestions > = {
+	allDomainSuggestions: [ MOCK_SUGGESTION ],
+	errorMessage: null,
+	state: DataStatus.Success,
+	retryRequest: jest.fn(),
+};
+
+jest.mock( '../src/hooks/use-domain-suggestions', () => ( {
+	useDomainSuggestions: () => mockUseDomainSuggestionsResult,
+} ) );
+
+beforeEach( () => {
+	jest.clearAllMocks();
+} );
+
+describe( 'Search for a domain and select a suggestion', () => {
+	it( 'should search for a domain', () => {
+		const { getByText } = render(
+			<DomainPicker
+				onDomainSelect={ jest.fn() }
+				analyticsUiAlgo="testalgo"
+				analyticsFlowId="12345"
+				initialDomainSearch="water"
+			/>
+		);
+
+		expect( getByText( '€15.00/year' ) ).toBeInTheDocument();
+	} );
+
+	it( 'should call the onDomainSelect callback when a button is clicked ', () => {
+		const onDomainSelectCallback = jest.fn();
+		const { getAllByText } = render(
+			<DomainPicker
+				onDomainSelect={ onDomainSelectCallback }
+				analyticsUiAlgo="testalgo"
+				analyticsFlowId="12345"
+				initialDomainSearch="water"
+				itemType="button"
+			/>
+		);
+
+		const selectDomainButtons = getAllByText( 'Select', { exact: true, selector: 'button' } );
+
+		// Expect as many buttons as there are suggestion items in the data
+		expect( selectDomainButtons ).toHaveLength(
+			mockUseDomainSuggestionsResult.allDomainSuggestions.length
+		);
+
+		// Check if callback is called correctly
+		fireEvent.click( selectDomainButtons[ 0 ] );
+
+		expect( onDomainSelectCallback ).toHaveBeenCalledTimes( 1 );
+		expect( onDomainSelectCallback ).toHaveBeenCalledWith(
+			mockUseDomainSuggestionsResult.allDomainSuggestions[ 0 ]
+		);
```

Comment:
We can replace these assertions with Jest's `mock` object.

```suggestion
	expect(onDomainSelectCallback.mock.calls).toEqual(mockUseDomainSuggestionsResult.allDomainSuggestions[ 0 ]);
```

Reply:
I think that the current code is much easier to read (and will output a much readable error when it fails).

On top of that, your suggestion doesn't ensure that `onDomainSelectCallback` gets called only once